src/server/common/helpers/wreck.js

```diff
@@ -0,0 +1,32 @@
+import Wreck from '@hapi/wreck'
+import { config } from '~/src/config/config.js'
+import { getTraceId } from '@defra/hapi-tracing'
+
+export const _wreck = Wreck.defaults({
+  events: true,
+  timeout: 3000,
+  baseUrl: config.get('fg_cw_backend_url')
+})
+
+_wreck.events.on('preRequest', (uri) => {
+  const traceId = getTraceId()
+
+  if (traceId) {
+    uri.headers[config.tracingHeader] = traceId
+  }
+})
+
+export const wreck = async (options = {}) => {
+  const { uri, method = 'GET', headers = {}, ...rest } = options
+  const reqOptions = {
+    ...rest,
+    headers: {
+      ...headers
+    }
+  }
+
+  const response = await _wreck.request(method, uri, reqOptions)
+  return _wreck.read(response, {
+    json: 'strict'
+  })
+}
```

Comment:
Could this not be simplified by using wreck direct? Would also get autocomplete.

```js
// wreck.js
import Wreck from '@hapi/wreck'
import { config } from '~/src/config/config.js'
import { getTraceId } from '@defra/hapi-tracing'

export const wreck = Wreck.defaults({
  events: true,
  timeout: 3000,
})

wreck.events.on('preRequest', (uri) => {
  const traceId = getTraceId()

  if (traceId) {
    uri.headers[config.tracingHeader] = traceId
  }
})
 
// somewhere.js
import { wreck } from './wreck.js'

const response = await wreck.post('https://some......', {
  json: true,
  payload: {
    foo: "bar"
  },
});

console.log(response.payload)
```